plugins/outputs/graylog/graylog_test.go

```diff
@@ -118,6 +118,10 @@ func TestWriteUDP(t *testing.T) {
 }
 
 func TestWriteTCP(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration test in short mode")
+	}
+
```

Comment:
Why do we need to skip this? I thought that access to the localhost can be seen as granted even in short-test mode...

Reply:
From the panic stack trace it isn't obvious why its timing out, there are some errors being ignored in the TCP server I could add checks for instead of skipping this test entirely and see if we can figure out the exact cause.

Reply:
well that didn't help: https://app.circleci.com/pipelines/github/influxdata/telegraf/8355/workflows/579f8fc0-cb5b-454a-873b-072966ce1f54/jobs/141621

Reply:
Probably isn't right to call `require.NoError` in the goroutine running the TCP server, need to rethink the error handling.

Reply:
Hmmm. There might be an issue due to reusing the fixed port in the test... I guess we should make this a random port and just use `net.Listen("tcp", "127.0.0.1:0")` here and avoid the strange `wg2` construct?

Reply:
@srebhan I've updated the test to use a random port as suggested and instead of using wg2 & wg3 I updated the test to wait for the errors so we can check them. What do you think? The test passed this time so that's nice but of course doesn't guarantee anything...

Reply:
@srebhan Are you happy with this? It looks ready to me